modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/api.ts

```diff
@@ -65,13 +66,13 @@ function getValidFields({
 			if (Liferay.FeatureFlags['LPS-186871']) {
 				fields.push({
 					children: getValidFields({
-						contextPath: `${contextPath}${propertyKey}.`,
+						contextPath: `${contextPath}${propertyKey}${NESTING_DELIMITER}`,
 						schemaName: propertyValue.$ref.replace(/^.*\//, ''),
 						schemas,
 					}),
 					label: propertyKey,
-					name: `${contextPath}${propertyKey}`,
-					type,
+					name: `${contextPath}${propertyKey}${NESTING_ROOT}`,
+					type: type ? type : 'object',
```

Comment:
This will fix the issue that prevents selecting root elements from *headless* endpoint. Not sure if it worth.

Reply:
We intentionally prevented selecting root elements from headless endpoint, it ended up in an error down the line. [Relevant Slack thread](https://liferay.slack.com/archives/C3JBR21HA/p1699957626114059). Is this an improvement on the fix, or the issue is no longer reproducible after this PR?

Reply:
I know that you apply a fix (https://github.com/liferay-frontend/liferay-portal/pull/3806/files#diff-8b819d07f50b442d5ec220fd20383c40b9713e088f455625482a2c6d9b8e9a3dL62) 
However, it is still reproducible. This is why I've created a bug -> https://liferay.slack.com/archives/C3JBR21HA/p1701252436956569

Reply:
In the video attached in the description, using `/headless-commerce-admin-catalog/v1.0/products` , root elements with children (catalog, diagram, ...) can be selected, but the root ones, like `catalog` ends up with the disabled button in the modal.

Reply:
If selection of this root element is compatible with DSM saving and FDS data resolution, it's more than fine to make this change :+1:  We do want to support the case in the end, filtering would be a temporary measure. It's a bit awkward to default type to `object`, but OpenAPI is full of awkward bits like this :sweat_smile: